### DISCUSSION TITLE
Bootstrap Forced Colors CSS for Windows High Contrast Themes

### DISCUSSION THREAD

**Discussion by coliff** [twbs/bootstrap#40607] [Show & Tell]
I've created a stylesheet by using the `forced-colors: active` media query which enhances the accessibility of Bootstrap 5 when using with [Hight Contrast themes in Windows](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/).

- GitHub repo: https://github.com/coliff/bootstrap-forced-colors-css
- npm: https://www.npmjs.com/package/bootstrap-forced-colors-css
- Demo: https://coliff.github.io/bootstrap-forced-colors-css/tests/

Some fixes made are a bit opinionated, e.g. increasing border width from 1px to 2px on form inputs to counter the fact that there is no box-shadow.

It'd be great to see some/most of these fixes be made to Bootstrap in the future. For now though, users can use this stylesheet to improve things for Windows users with High Contrast themes.

```html
<link
  rel="stylesheet"
  href="https://cdn.jsdelivr.net/npm/bootstrap-forced-colors-css@1/css/bootstrap-forced-colors.min.css" 
  media="screen and (forced-colors: active)">
```

<details>
<summary>How to test with Forced Colors Mode Emulation in Chrome / Edge</summary>

![edge-test-forced-colors-active](https://github.com/twbs/bootstrap/assets/1212885/dc681871-7605-42d8-82d0-b6361e7f3faa)

![edge-test-forced-colors-active-light-mode](https://github.com/twbs/bootstrap/assets/1212885/df7766a8-d825-4845-8a2e-1f733dbd38e8)

</details>

### Interesting links about Windows High Contrast Mode:

- [Smashing Magazine article](https://www.smashingmagazine.com/2022/03/windows-high-contrast-colors-mode-css-custom-properties/)
- [CanIUse Support](https://caniuse.com/mdn-css_at-rules_media_forced-colors) (Note that currently it's Windows-only)
- [WHCM and System Colors](https://adrianroselli.com/2021/02/whcm-and-system-colors.html)

**Comment by julien-deramond** [comment]
Hey @coliff!
Sounds really cool and interesting, good job! 🎉
You've already mentioned it in [this GitHub issue comment](https://github.com/twbs/bootstrap/issues/35941#issuecomment-2209277311) 👌, so we'll definitely consider it while developing the high contrast mode (hopefully for v6).

> Some fixes made are a bit opinionated, e.g. increasing border width from 1px to 2px on form inputs to counter the fact that there is no box-shadow.
> It'd be great to see some/most of these fixes be made to Bootstrap in the future.

I'll add that to my to-do list. Maybe some of the changes you've made can be integrated into v5 already.
